model_signing/signature/signature.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This package provides the functionality to sign models"""
+import abc
+
+from in_toto_attestation.v1 import statement
+from sigstore_protobuf_specs.dev.sigstore.bundle import v1 as bundle_pb
+
+
+class VerificationMaterial(metaclass=abc.ABCMeta):
+    """Generic verification material"""
+
+
+class SigstoreVerificationMaterial(VerificationMaterial):
+    """
+    SigstoreVerificationMaterial supports to conversion to the sigstore
+    verification material message.
+    """
+
+    @abc.abstractmethod
+    def to_sigstore_verification_material(
+            self) -> bundle_pb.VerificationMaterial:
+        """
+        Converts the verification material to the sigstore verification
+        material message.
+
+        Returns:
+            bundle_pb.VerificationMaterial: Sigstore verification material 
+        """
+
+
+class BytesSigner(abc.ABC):
+    """BytesSigner is the abstract base class for signing bytes."""
+
+    @abc.abstractmethod
+    def sign(self, data: bytes) -> bytes:
```

Comment:
This is the same as [`signing.Signer`](https://github.com/sigstore/model-transparency/blob/2198c9d471be3058e2d2fced7e6ddb5d3c116b5f/model_signing/signing/signing.py#L102). I'm curious why we'd duplicate the class hierarchy instead of using it?